src/Akka.Persistence.MongoDb/Snapshot/MongoDbGridFSSnapshotStore.cs

```diff
@@ -0,0 +1,356 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="MongoDbSnapshotStore.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2016 Akka.NET project <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Akka.Configuration;
+using Akka.Event;
+using Akka.Persistence.Snapshot;
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization;
+using MongoDB.Driver;
+using MongoDB.Driver.GridFS;
+using MongoDB.Driver.Linq;
+
+#nullable enable
+namespace Akka.Persistence.MongoDb.Snapshot;
+
+/// <summary>
+/// A large snapshot SnapshotStore implementation for writing snapshots to MongoDB.
+/// Implements MongoDB GridFS storage mechanisms to support snapshots larger than 16 megabytes
+/// </summary>
+public class MongoDbGridFsSnapshotStore : SnapshotStore
+{
+    private const string PersistenceIdKey = "_pid";
+    private const string SequenceNrKey = "_snr";
+    private const string TimestampKey = "_ts";
+    
+    private static readonly ClientSessionOptions EmptySessionOptions = new();
+        
+    private readonly MongoDbSnapshotSettings _settings;
+    private readonly GridFSBucketOptions _bucketOptions;
+    // ReSharper disable InconsistentNaming
+    private IMongoDatabase? _mongoDatabase_DoNotUseDirectly;
+    private GridFSBucket? _snapshotGridFSBucket_DoNotUseDirectly;
+    // ReSharper enable InconsistentNaming
+
+    /// <summary>
+    /// Used to cancel all outstanding commands when the actor is stopped.
+    /// </summary>
+    private readonly CancellationTokenSource _pendingCommandsCancellation = new();
+
+    private readonly Akka.Serialization.Serialization _serialization;
+    private readonly ILoggingAdapter _log;
+
+    public MongoDbGridFsSnapshotStore() : this(MongoDbPersistence.Get(Context.System).SnapshotStoreSettings)
+    {
+    }
+
+    public MongoDbGridFsSnapshotStore(Config config) : this(new MongoDbSnapshotSettings(config))
+    {
+    }
+
+    public MongoDbGridFsSnapshotStore(MongoDbSnapshotSettings settings)
+    {
+        _settings = settings;
+        _serialization = Context.System.Serialization;
+        _bucketOptions = new GridFSBucketOptions
+        {
+            //ReadConcern = settings.Transaction ? ReadConcern.Snapshot : ReadConcern.Default,
```

Comment:
Are these settings not supported with `GridFsBucketOptions` ?

Reply:
I might need to change that because we're only using the GridFS built-in features to save a snapshot, still need to compare the supported transaction models with the rest of the code (load/delete)